lib/broadway_rabbitmq/producer.ex

```diff
@@ -3,13 +3,15 @@ defmodule BroadwayRabbitmq.Producer do
 
   require Logger
 
-  alias Broadway.{Message, Acknowledger}
+  alias Broadway.{Message, Acknowledger, Producer}
   alias BroadwayRabbitmq.Backoff
 
   @behaviour Acknowledger
+  @behaviour Producer
 
   @impl true
   def init(opts) do
+    Process.flag(:trap_exit, true)
     client = opts[:client] || BroadwayRabbitmq.AmqpClient
```

Comment:
Do we really want to expose this as an abstraction or is it private for now?

Reply:
It's private. I have no intention to make it public.